model/lifecycle.go

```diff
@@ -1702,19 +1702,29 @@ func addNewTasks(ctx context.Context, creationInfo TaskCreationInfo, existingBui
 		// Build a list of tasks that haven't been created yet for the given variant, but have
 		// a record in the TVPairSet indicating that it should exist
 		tasksToAdd := []string{}
+		existingTasksToActivate := []task.Task{}
 		for _, taskName := range creationInfo.Pairs.ExecTasks.TaskNames(b.BuildVariant) {
-			if ok := existingTasksIndex[taskName]; ok {
+			if t, ok := existingTasksIndex[taskName]; ok {
+				if !t.Activated {
+					existingTasksToActivate = append(existingTasksToActivate, t)
+				}
 				continue
 			}
 			tasksToAdd = append(tasksToAdd, taskName)
 		}
 		displayTasksToAdd := []string{}
 		for _, taskName := range creationInfo.Pairs.DisplayTasks.TaskNames(b.BuildVariant) {
-			if ok := existingTasksIndex[taskName]; ok {
+			if t, ok := existingTasksIndex[taskName]; ok {
+				if !t.Activated {
+					existingTasksToActivate = append(existingTasksToActivate, t)
+				}
 				continue
 			}
 			displayTasksToAdd = append(displayTasksToAdd, taskName)
 		}
+		if err = SetActiveState(evergreen.DefaultTaskActivator, true, existingTasksToActivate...); err != nil {
```

Comment:
We should only call this if existingTasksToActivate isn't empty, right? Also would we then need to add this build to buildIdsToActivate?

Reply:
I'm slightly wondering if this would be less edge-casey if we just made a new function activateExistingTasks that the UI calls, rather than putting this into addNewTasks. I don't think it would cost so much more to rebuild the existing tasks index. That being said this is also fine if you prefer it and/or if we need this for all places addNewTasks is called (like I see we tested using the generate tasks job); we should update the function comment though and add a couple of comments here.

Reply:
`SetActiveState` would noop with an empty task list input, but to be more explicit I put it behind a conditional.  It also handles build activation [on its own](https://gist.github.com/hadjri/23262276c9ace0f05efe5ee3e804272c#file-task_lifecycle-go-L106) so I don't think the second part is necessary.

Really the only function that needs this behavior is `ConfigurePatch`, not the task generation process, so I moved this logic into a new helper as you suggested and propagated the call up one level into `addNewTasksAndBuildsForPatch` for cleanliness purposes.